lib/components/ButtonGroupView.jsx

```diff
@@ -0,0 +1,152 @@
+    let mruMenuItems;
+    if (mruFiles.length) {
+        mruMenuItems = mruFiles.map(filePath => (
+            <MenuItem key={filePath} onSelect={() => openFile(filePath)}>{filePath}</MenuItem>
+        ));
+    } else {
+        mruMenuItems = (<MenuItem disabled>No recently used files</MenuItem>);
+    }
+    return mruMenuItems;
+};
+
+const ButtonGroupView = props => {
+    const {
+        openFile,
+        closeFiles,
+        refreshAllFiles,
+        openFileDialog,
+        onToggleFileList,
+        performWrite,
+        performRecover,
+        performRecoverAndWrite,
+        mruFiles,
+        targetIsReady,
+        targetIsWritable,
+        targetIsRecoverable,
+    } = props;
+    const mruMenuItems = MruMenuItems(mruFiles, openFile);
+
+    return (
+        <div>
+            <Dropdown pullRight id="files-dropdown">
+                <Dropdown.Toggle onClick={onToggleFileList}>
+                    <Glyphicon glyph="folder-open" />Add a .hex file
+                </Dropdown.Toggle>
+                <Dropdown.Menu>
+                    { mruMenuItems }
+                    <MenuItem divider />
+                    <MenuItem onSelect={openFileDialog}>Browse...</MenuItem>
+                </Dropdown.Menu>
+            </Dropdown>
+            <Button onClick={refreshAllFiles}>
+                <Glyphicon glyph="refresh" />Reload .hex files
+            </Button>
+            <Button onClick={closeFiles}>
+                <Glyphicon glyph="minus-sign" />Clear files
+            </Button>
+
+            <hr style={{ borderColor: 'transparent', margin: '5px 0px' }} />
+
+            <Button
+                onClick={performRecover}
+                disabled={!targetIsReady || !targetIsRecoverable}
+            >
+                <Glyphicon glyph="remove-sign" />Erase all
+            </Button>
+            <Button
+                onClick={performWrite}
+                disabled={!targetIsReady || !targetIsWritable}
+            >
+                <Glyphicon glyph="download-alt" />Write
+            </Button>
+            <Button
+                onClick={performRecoverAndWrite}
+                disabled={!targetIsReady || !targetIsRecoverable}
+            >
+                <Glyphicon glyph="save" />Erase all & write
+            </Button>
+        </div>
+    );
+};
+
+ButtonGroupView.propTypes = {
+    openFile: PropTypes.func.isRequired,
+    closeFiles: PropTypes.func.isRequired,
+    refreshAllFiles: PropTypes.func.isRequired,
+    onToggleFileList: PropTypes.func.isRequired,
+    openFileDialog: PropTypes.func.isRequired,
+    performWrite: PropTypes.func.isRequired,
+    performRecover: PropTypes.func.isRequired,
+    performRecoverAndWrite: PropTypes.func.isRequired,
+    mruFiles: PropTypes.arrayOf(PropTypes.string).isRequired,
+    targetIsReady: PropTypes.bool.isRequired,
+    targetIsWritable: PropTypes.bool.isRequired,
+    targetIsRecoverable: PropTypes.bool.isRequired,
+};
+
+export default connect(
```

Comment:
Move this to _containers/buttonGroupView.js_

Reply:
Moved.